### ISSUE TITLE
[BUG/ISSUE] build_mechanism.sh not working

### ISSUE THREAD

**Issue by cdholmes** [geoschem/geos-chem#1372]
### What institution are you from?
Florida State U

### Description of the problem
I am unable to compile the chemical mechanism with KPP in GCClassic 13.4. The build_mechanism.sh has a few bugs that I solved and one that I haven't.

### GEOS-Chem version
GCClassic 13.4.0
KPP 3.0.0_rc.2 and KPP 2.6 
No modifications to either

### Description of troubleshooting performed
The following issues arose with both KPP2.6 and 3.0.0_rc.2. I get the same results regardless of which mechanism I try to build (custom, fullchem, etc).
1. build_mechanism.sh first told me "ERROR: You must use KPP version 2.3.0_gc or later!". I am using a newer KPP version, but the script checks the KPP version by looking at the directory name. Running the KPP command itself would be a better way to get the KPP version. I fixed this by commenting out this check.
2. I then got the error, "KPP failed to build 'gckpp_Rates.F90'! Aborting." This is because kpp generates files ending .f90 but the build mechanism expects .F90. I fixed this by adding a command "rename f90 F90 *f90" to build_mechanism.sh. This isn't command won't work on all systems but it would be nice if the build script could handle the case change. 
3. KPP now runs, but the OHreact_parser.py generates the following 
```
Traceback (most recent call last):
  File "../OHreact_parser.py", line 93, in <module>
    rn = int(spl[1].split('index ')[1])
ValueError: invalid literal for int() with base 10: 'up to 30'
```
The cause is in gckpp_Monitor.F90. The last index in every block of equations has an extra "up to N" as in the following.
```
     '            2 MO2 --> 2 CH2O + 2 HO2                                                                ', & ! index 28
     '          MP + OH --> MO2 + H2O                                                                     ', & ! index 29
     '          MP + OH --> CH2O + OH + H2O                                                               ' /) ! index up to 30
```
It's not clear if KPP or OHreact_parser.py should be modified to address this behavior.

### Software versions
- CMake version: 3.20.2 (probably not relevant)
- Compilers: KPP compiled with gcc 11.2

### Log files
Log from build_mechanism.sh
```
$ ./build_mechanism.sh custom

This is KPP-3.0.0-rc.2.

KPP is parsing the equation file.
KPP is computing Jacobian sparsity structure.
KPP is starting the code generation.
KPP is initializing the code generation.
KPP is generating the monitor data:
    - gckpp_Monitor
KPP is generating the utility data:
    - gckpp_Util
KPP is generating the global declarations:
    - gckpp_Main
KPP is generating the ODE function:
    - gckpp_Function
KPP is generating the ODE Jacobian:
    - gckpp_Jacobian
    - gckpp_JacobianSP
KPP is generating the linear algebra routines:
    - gckpp_LinearAlgebra
KPP is generating the utility functions:
    - gckpp_Util
KPP is generating the rate laws:
    - gckpp_Rates
KPP is generating the parameters:
    - gckpp_Parameters
KPP is generating the global data:
    - gckpp_Global
KPP is generating the driver from none.f90:
    - gckpp_Main
KPP is starting the code post-processing.

KPP has successfully created the model "gckpp".

Traceback (most recent call last):
  File "../OHreact_parser.py", line 93, in <module>
    rn = int(spl[1].split('index ')[1])
ValueError: invalid literal for int() with base 10: 'up to 30'
```

gckpp_Monitor.F90 attached here, but renamed .txt for github support
[gckpp_Monitor.F90.txt](https://github.com/geoschem/geos-chem/files/9560409/gckpp_Monitor.F90.txt)

**Comment by yantosca** [comment]
Thanks for writing @cdholmes.  If you are using GEOS-Chem 13.4.1 with KPP 2.6.0, you'll need to replace the files `KPP/build_mechanism.sh` and `KPP/OHreact_parser.py` with the newer versions of these files from the geoschem/geos-chem dev branch.  See the updated files at:

  - https://github.com/geoschem/geos-chem/blob/dev/KPP/build_mechanism.sh
  - https://github.com/geoschem/geos-chem/blob/dev/KPP/OHreact_parser.py

These newer files will be included in GEOS-Chem 14.0.0 (to be released imminently).  So KPP 2.5.0+ should work out-of-the-box in GEOS-Chem 14.0.0 and later versions.

Also note, we have confirmed that the KPP 2.5.0+ does results in identical solver code being generated to the older KPP 2.3.3_gc version.  All of the updates were on the front-end rather than in the algorithm used to generate the F90 solver code.

**Comment by cdholmes** [comment]
Thank you @yantosca. I updated the files that you mentioned and I can build the 13.4.1 mechanism with KPP 2.6... almost

The build_mechanism.sh script expects files to end with F90, but kpp produces files ending with f90 (lower case). The rename command easily fixes this, but I don't think it has consistent syntax across *nix OSes.
